File: sdks/csharp/DiscoveryPipeline.cs
using Vectorizer.Models;

namespace Vectorizer;

public partial class VectorizerClient
{
    /// <summary>
    /// Runs a multi-query broad search across all collections.
    /// POST /discovery/broad_discovery
    /// </summary>
    public async Task<BroadDiscoveryResponse> BroadDiscoveryAsync(
        BroadDiscoveryRequest request,
        CancellationToken cancellationToken = default)
    {
        return await RequestAsync<BroadDiscoveryResponse>(
            "POST", "/discovery/broad_discovery", request, cancellationToken);
    }

    /// <summary>
    /// Runs a focused semantic search within a single collection.
    /// POST /discovery/semantic_focus
    /// </summary>
    public async Task<SemanticFocusResponse> SemanticFocusAsync(
        SemanticFocusRequest request,
        CancellationToken cancellationToken = default)
    {
        return await RequestAsync<SemanticFocusResponse>(
            "POST", "/discovery/semantic_focus", request, cancellationToken);
    }

    /// <summary>
    /// Promotes README-quality chunks to the top of a result set.
    /// POST /discovery/promote_readme
    /// </summary>
    public async Task<PromoteReadmeResponse> PromoteReadmeAsync(
        PromoteReadmeRequest request,
        CancellationToken cancellationToken = default)
    {
        return await RequestAsync<PromoteReadmeResponse>(
            "POST", "/discovery/promote_readme", request, cancellationToken);
    }

    /// <summary>
    /// Compresses a chunk set into a concise bullet list.
    /// POST /discovery/compress_evidence
    /// </summary>
    public async Task<CompressEvidenceResponse> CompressEvidenceAsync(
        CompressEvidenceRequest request,
        CancellationToken cancellationToken = default)
    {
        return await RequestAsync<CompressEvidenceResponse>(
            "POST", "/discovery/compress_evidence", request, cancellationToken);
    }

    /// <summary>
    /// Organises bullets into a structured answer plan.
    /// POST /discovery/build_answer_plan
    /// </summary>
    public async Task<AnswerPlan> BuildAnswerPlanAsync(
        AnswerPlanRequest request,
        CancellationToken cancellationToken = default)
    {
        return await RequestAsync<AnswerPlan>(
            "POST", "/discovery/build_answer_plan", request, cancellationToken);
    }

    /// <summary>
    /// Renders an answer plan into a final LLM prompt string.
    /// POST /discovery/render_llm_prompt
    /// </summary>
    public async Task<LlmPrompt> RenderLlmPromptAsync(
        RenderPromptRequest request,
        CancellationToken cancellationToken = default)
    {
        return await RequestAsync<LlmPrompt>(
            "POST", "/discovery/render_llm_prompt", request, cancellationToken);
    }
}

File: sdks/go/discovery_pipeline.go
package vectorizer

// BroadDiscovery runs a multi-query broad search across all collections.
//
// POST /discovery/broad_discovery
func (c *Client) BroadDiscovery(req *BroadDiscoveryRequest) (*BroadDiscoveryResponse, error) {
	var result BroadDiscoveryResponse
	if err := c.request("POST", "/discovery/broad_discovery", req, &result); err != nil {
		return nil, err
	}
	return &result, nil
}

// SemanticFocus runs a focused semantic search within a single collection.
//
// POST /discovery/semantic_focus
func (c *Client) SemanticFocus(req *SemanticFocusRequest) (*SemanticFocusResponse, error) {
	var result SemanticFocusResponse
	if err := c.request("POST", "/discovery/semantic_focus", req, &result); err != nil {
		return nil, err
	}
	return &result, nil
}

// PromoteReadme promotes README-quality chunks to the top of a result set.
//
// POST /discovery/promote_readme
func (c *Client) PromoteReadme(req *PromoteReadmeRequest) (*PromoteReadmeResponse, error) {
	var result PromoteReadmeResponse
	if err := c.request("POST", "/discovery/promote_readme", req, &result); err != nil {
		return nil, err
	}
	return &result, nil
}

// CompressEvidence compresses a chunk set into a concise bullet list.
//
// POST /discovery/compress_evidence
func (c *Client) CompressEvidence(req *CompressEvidenceRequest) (*CompressEvidenceResponse, error) {
	var result CompressEvidenceResponse
	if err := c.request("POST", "/discovery/compress_evidence", req, &result); err != nil {
		return nil, err
	}
	return &result, nil
}

// BuildAnswerPlan organises bullets into a structured answer plan.
//
// POST /discovery/build_answer_plan
func (c *Client) BuildAnswerPlan(req *AnswerPlanRequest) (*AnswerPlan, error) {
	var result AnswerPlan
	if err := c.request("POST", "/discovery/build_answer_plan", req, &result); err != nil {
		return nil, err
	}
	return &result, nil
}

// RenderLlmPrompt renders an answer plan into a final LLM prompt string.
//
// POST /discovery/render_llm_prompt
func (c *Client) RenderLlmPrompt(req *RenderPromptRequest) (*LlmPrompt, error) {
	var result LlmPrompt
	if err := c.request("POST", "/discovery/render_llm_prompt", req, &result); err != nil {
		return nil, err
	}
	return &result, nil
}
